include/dgl/immutable_graph.h

```diff
@@ -32,6 +32,30 @@ class ImmutableGraph: public GraphInterface {
     dgl_id_t edge_id;
   };
 
+  // Edge list indexed by edge id;
+  struct EdgeList {
+    typedef std::shared_ptr<EdgeList> Ptr;
+    std::vector<dgl_id_t> src_points;
+    std::vector<dgl_id_t> dst_points;
+
+    EdgeList(int64_t len, dgl_id_t val) {
+      src_points.resize(len, val);
+      dst_points.resize(len, val);
+    }
+
+    void register_edge(dgl_id_t eid, dgl_id_t src, dgl_id_t dst) {
+      CHECK_LT(eid, src_points.size()) << "Invalid edge id " << eid;
+      src_points[eid] = src;
+      dst_points[eid] = dst;
+    }
+
+    static EdgeList::Ptr FromCSR(
+        std::vector<int64_t> *indptr,
+        std::vector<dgl_id_t> *indices,
+        std::vector<dgl_id_t> *edge_ids,
+        bool in_csr);
```

Comment:
can you make the input arguments const reference? input pointers usually mean the vectors will be modified by the method.

Reply:
updated